src/Adapter/Image/Uploader/CategoryCoverImageUploader.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use Category;
+use ImageManager;
+use ImageType;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageOptimizationException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\MemoryLimitException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Class CategoryCoverImageUploader.
+ *
+ * @internal
+ */
+final class CategoryCoverImageUploader extends AbstractImageUploader
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws MemoryLimitException
+     * @throws ImageOptimizationException
+     * @throws ImageUploadException
+     * @throws UploadedImageConstraintException
+     */
+    public function upload($id, UploadedFile $uploadedImage)
+    {
+        $this->deleteOldImage($id);
+        $this->checkImageIsAllowedForUpload($uploadedImage);
```

Comment:
even though i agree with you, but lets keep behavior as it was in legacy. we can change it in another PR if it must be. :)

Reply:
well when there is a risk to break something or generating side effects, yeah I would agree But here clearly it was not thought through in the legacy code, and it won't break a desired behavior it will just fix a potential bug
Refacto/migation is not only about doing the same shit in a classier way :P If we can fix small bugs like this one in the process we should

Reply:
> and it won't break a desired behavior it will just fix a potential bug

i've tested it and apparently we cant delele old images after uploading new one. The reason is that both new and old images use same naming, e.g. `{categoryId}-thumb.jpg`. so deleting old ones would delete actually new ones. :P

Reply:
😱

Reply:
damn!!! well too bad ^^